### ISSUE TITLE
SSIM on tensorflow

### ISSUE THREAD

**Issue by BenjaminDug** [BenjaminDug/Help#2]
Hello, 

I tried to implement the SSIM (structural similarity index metric) in Tensorflow. I wrote the following code:

`def ssim_tensorflow(x_true,x_predict,N):
    
    #N is the size of the image
    k1=tf.constant(0.01)
    k2=tf.constant(0.03)
    L=2**8
    C1=(k1*L)**2
    C2=(k2*L)**2
    
    #For sigxy
    mean_true,_=tf.nn.moments(x_true,[1,2,3],keep_dims=True)
    mean_predict,_= tf.nn.moments(x_predict,[1,2,3],keep_dims=True)
    
    #For value calculation
    u_true,var_true=tf.nn.moments(x_true,[1,2,3])
    u_predict,var_predict = tf.nn.moments(x_predict,[1,2,3])    
    
    xy=(x_true-mean_true)*(x_predict-mean_predict)

    sigxy=tf.reduce_sum(xy,axis=[1,2,3])/N
    
    nomi= ((2*u_true*u_predict+C1)*(2*sigxy+C2))
    deno=((u_true**2+u_predict**2+C1)*(var_true+var_predict+C2))
    
    value=nomi/deno
    
    return value`

I wanted to validate the function thanks to "compare_ssim" which is part of skimage.measure.

When the 2 images are equal, I have the good result: 1.0 , but when they are different I do not have the same result. I don't find where is the problem. 

Thank you for your help :)
